openwisp_controller/config/base/who_is.py

```diff
@@ -0,0 +1,106 @@
+from ipaddress import ip_address, ip_network
+
+from django.core.cache import cache
+from django.core.exceptions import ValidationError
+from django.db import models, transaction
+from django.utils.translation import gettext_lazy as _
+from jsonfield import JSONField
+
+from openwisp_utils.base import TimeStampedEditableModel
+
+from ..who_is.service import WhoIsService
+
+
+class AbstractWhoIsInfo(TimeStampedEditableModel):
+    """
+    Abstract model to store WhoIs information
+    for a device.
+    """
+
+    id = None
+    ip_address = models.GenericIPAddressField(db_index=True, primary_key=True)
+    isp = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text=_("Organization associated with registered Autonomous System Number"),
+    )
+    asn = models.CharField(
+        max_length=6,
+        blank=True,
+        help_text=_("Autonomous System Number"),
+    )
+    timezone = models.CharField(
+        max_length=35,
+        blank=True,
+        help_text=_("Time zone"),
+    )
+    address = JSONField(
+        default=dict,
+        help_text=_("Address"),
+        blank=True,
+    )
+    cidr = models.CharField(
+        max_length=20,
+        blank=True,
+        help_text=_("CIDR"),
+    )
+
+    class Meta:
+        abstract = True
+
+    def clean(self):
+        if ip_address(self.ip_address).is_private:
+            raise ValidationError(
+                _("WhoIs information cannot be created for private IP addresses.")
```

Comment:
is there a test which raises this on purpose and verified the error message matches this? I can't find anything like that, make sure it's included.

Reply:
There is a test which raises this, i will add the code which matches the message as well.